tests/baselines/reference/intersectionNarrowing.types

```diff
@@ -59,11 +59,11 @@ function f4<T>(x: T & 1 | T & 2) {
 
         case 1: x; break;  // T & 1
 >1 : 1
->x : T & 1
+>x : (T & 1) | (T & 2)
 
         case 2: x; break;  // T & 2
 >2 : 2
->x : T & 2
+>x : (T & 1) | (T & 2)

```

Comment:
Fallout from permissive comparability: We use comparability in switch-case narrowing (in `narrowTypeBySwitchOnDiscriminant`) and since `T` is unconstrained and thus extends `unknown`, both the `1` and `2` intersections are considered comparable with both `1` and `2` because of the `T` part. Some extra work could possibly be done to prevent this by changing how comparability works over intersections, probably.

Reply:
(It's likely a similar change occurs in narrowing assignments, since we also use bidirectional comparability there, and maybe in other places more subtly since we use unidirectional comparability in `narrowTypeByDiscriminant`)